### ISSUE TITLE
[proposal] Public `VecNormalize._normalize_observation`

### ISSUE THREAD

**Issue by shwang** [hill-a/stable-baselines#602]
I have saved demonstrations from a particular environment, and also a PPO2 policy trained on a normalized version of the same environment.

To properly compare the unnormalized observations in the demonstration against the PPO2 policy I am using `VecNormalize._normalize_observation(obs_demo)`. For ease of use, I'm wondering if the maintainers would be fine with:

1. Rename `_normalize_obs` to `normalize_obs` so that it is a "public" method.
2. Add `training` param to `normalize_obs()` so that we can call the `normalize_obs` without changing normalizing statistics. `VecNormalize.step()` would call `self.normalize_obs(obs, training=self.training)`.

**Comment by araffin** [comment]
Hello,

I'm totally for that feature, in fact it would good to prepare for https://github.com/hill-a/stable-baselines/issues/200

I would rather decouple the update from the normalization to avoid confusion (so no `training` parameter is needed)

**Comment by AdamGleave** [comment]
While we're at it, how about exposing the reward normalization, which currently is embedded in `step_wait`?

Combining this and @araffin's proposals, we'd have two methods `normalize_observation` and `normalize_reward`, both of which are side-effect free. Then inside `step_wait` there is the `self.training` check which updates `ret_rms` and `obs_rms`, and `reset` has a similar check but updates just `obs_rms`.

**Comment by araffin** [comment]
Good point, i think this is also a good moment to rename `rew` to `reward`

**Comment by shwang** [comment]
> Good point, i think this is also a good moment to rename `rew` to `reward`

@araffin Did you mean that you want to rename `self.get_original_{rew=>reward}`?

**Comment by araffin** [comment]
> > Good point, i think this is also a good moment to rename `rew` to `reward`
> 
> @araffin Did you mean that you want to rename `self.get_original_{rew=>rewards}`? (If so I will also `self.get_original_{obs=>observations}`

only the first one (the 2nd one would break some code and is maybe too long), and it would be singular i think (but it does not really matter)

**Comment by madhekar** [comment]
normalize_obs saturates to clip 10, not sure it's useful => although get_original_obs() works. that makes it useful only with virtual /  toy environments.
